packages/luvi/build.sh

```diff
@@ -2,36 +2,40 @@ TERMUX_PKG_HOMEPAGE=https://luvit.io
 TERMUX_PKG_DESCRIPTION="A project in-between luv and luvit."
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="Komo @mbekkomo"
-TERMUX_PKG_VERSION=20241009
-_commit=32b274bed2a291068eda6cf7a296328b86d682a9
-_version=2.14.0+g${_commit::7}
+TERMUX_PKG_VERSION=2.15.0
+TERMUX_PKG_SRCURL=git+https://github.com/luvit/luvi
 TERMUX_PKG_GIT_BRANCH=master
-TERMUX_PKG_SRCURL=git+https://github.com/luvit/luvi.git
-TERMUX_PKG_DEPENDS="pcre2, openssl, luv, libluajit"
+TERMUX_PKG_DEPENDS="pcre2, openssl, luv, lua51-lpeg, zlib, libluajit"
 TERMUX_PKG_SUGGESTS="lit, luvit"
 TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 	-DWithSharedLibluv=On
 	-DWithOpenSSL=On
 	-DWithSharedOpenSSL=On
 	-DWithPCRE2=On
 	-DWithSharedPCRE2=On
 	-DWithLPEG=On
+	-DWithSharedLPEG=On
+	-DWithZLIB=On
+	-DWithSharedZLIB=ON
 	-DLIBLUV_INCLUDE_DIR=${TERMUX_PREFIX}/include/luv
-	-DLIBLUV_LIBRARIES=${TERMUX_PREFIX}/lib/libluv.so
 	-DLUAJIT_INCLUDE_DIR=${TERMUX_PREFIX}/include/luajit-2.1
-	-DLUAJIT_LIBRARIES=${TERMUX_PREFIX}/lib/libluajit.so
 	-DLIBUV_INCLUDE_DIR=${TERMUX_PREFIX}/include
-	-DLIBUV_LIBRARIES=${TERMUX_PREFIX}/lib/libuv.so
 	-DOPENSSL_INCLUDE_DIR=${TERMUX_PREFIX}/include
-	-DOPENSSL_LIBRARIES=${TERMUX_PREFIX}/lib
 	-DPCRE2_INCLUDE_DIR=${TERMUX_PREFIX}/include
-	-DPCRE2_LIBRARIES=${TERMUX_PREFIX}/lib
+	-DZLIB_INCLUDE_DIR=${TERMUX_PREFIX}/include
+	-DLIBLUV_LIBRARIES=${TERMUX_PREFIX}/lib/libluv.so
+	-DLUAJIT_LIBRARIES=${TERMUX_PREFIX}/lib/libluajit.so
+	-DLIBUV_LIBRARIES=${TERMUX_PREFIX}/lib/libuv.so
+	-DOPENSSL_LIBRARIES=${TERMUX_PREFIX}/lib/libssl.so;${TERMUX_PREFIX}/lib/libcrypto.so
+	-DPCRE2_LIBRARIES=${TERMUX_PREFIX}/lib/libpcre2-8.so
+	-DZLIB_LIBRARIES=${TERMUX_PREFIX}/lib/libz.so
+	-DLPEG_LIBRARIES=${TERMUX_PREFIX}/lib/liblpeg-5.1.so
 "
 
-termux_step_post_get_source() {
-	git fetch --unshallow
-	git checkout "${_commit}"
+termux_step_get_source() {
+	git clone --recursive "${TERMUX_PKG_SRCURL/git+/}" "${TERMUX_PKG_SRCDIR}"
```

Comment:
Doesn't the normal `git+` SRCURL code path for getting the source code do a recursive clone?

Reply:
The git command returns command not found error followed by the message unknown `remote-git-https` command or something

Reply:
I'd like to know the exact "or something".

Reply:
The standard `termux_git_clone_src()` code path should do a full incremental recursive clone.
https://github.com/termux/termux-packages/blob/0e462e498b9f6327b748f94a3f5f145ef38c764c/scripts/build/get_source/termux_git_clone_src.sh#L18-L36